File: fibonacci/fibonacci.rb
#! /usr/bin/env ruby

def fib(rot)
  
  old_value = 1
  new_value = 0

  while rot > 0 do
    old_value, new_value = new_value, new_value + old_value
    rot -= 1
  end

  return new_value
  
end

# If no arguments, run 20 fibs:
if ARGV.length == 0 then
  (0...10).each {|i| puts "#{fib(i)}\t#{fib(i + 10)}"}

# Else, fib the first argument:
else
  puts fib(ARGV[0].to_i)

end

=begin
Name: Fibonacci Sequence
Language: Ruby
State: Done

Prints out numbers from the fibonacci sequence


Example: ./fibonacci.rb
Example2: ./fibonacci.rb 42
=end

File: fibonacci/fibonacci.py
#! /usr/bin/env python3

from sys import argv

def fib(rot):
    
    old_value = 1
    new_value = 0

    while rot > 0:
        
        old_value, new_value = new_value, new_value + old_value
        rot -= 1

    return new_value

# If no arguments, print 20 fibs::
if len(argv) == 1:
    for r in range(10):
        print("%s\t%s" % (fib(r), fib(r+10)))

# Otherwise we'll fib the first argument:
else:
    print(fib(int(argv[1])))

""" TAIL INFO:
Name: Fibonacci Sequence
Language: Python3
State: Done

Prints out numbers from the fibonacci sequence


Example: ./fibonacci.py
Example2: ./fibonacci.py 42
"""
